server/src/termdb.survival.js

```diff
@@ -44,8 +44,13 @@ export async function get_survival(q, ds) {
 		const st = q[`term${survTermIndex}`]
 		// ot: overlay term, the series term
 		const ot = q[`term${survTermIndex == 1 ? 2 : 1}`]
-
-		const data = await getData({ terms: twLst, filter: q.filter }, ds, q.genome, true)
+		const types = new Set()
+		const ids = [st.id]
+		if (ot) ids.push(ot.id)
+		if (q.term0) ids.push(q.term0.id)
+		for (const id of ids) types.add(ds.cohort.termdb.term2SampleType.get(id))
+		const onlyChildren = types.size > 1
```

Comment:
i would like to see more comments on these logic. especially where an assumption is made (e.g. survival data is always on "root" samples). so that we can be aware of those. thanks a lot

Reply:
there is no assumption that survival is a root term